PROTO_tests/tests/setops_test.py

```diff
@@ -0,0 +1,783 @@
+            a = a + 2**200
+            b = np.array([-1, 2, 2, 3]).astype(ak.uint64)
+            b = b + 2 ** 200
+        elif dtype == ak.bool:
+            a = np.array([True, False, False, True]).astype(dtype)
+            b = np.array([True, True, False, False]).astype(dtype)
+        else:
+            a = b = None
+        return a, b
+
+    @staticmethod
+    def make_np_arrays_multi_small(dtype):
+        if dtype == ak.int64 or dtype == ak.uint64:
+            a = np.array([-1, -3, 0, 1, 2, 3]).astype(dtype)
+            b = np.array([-1, 0, 0, 7, 8, 3]).astype(dtype)
+
+            c = np.array([-1, -11, 0, 4, 5, 3]).astype(dtype)
+            d = np.array([-1, -4, 0, 7, 8, 3]).astype(dtype)
+        elif dtype == ak.bigint:
+            a = np.array([-1, -3, 0, 1, 2, 3]).astype(ak.uint64)
+            a = a + 2**200
+            b = np.array([-1, 0, 0, 7, 8, 3]).astype(ak.uint64)
+            b = b + 2**200
+
+            c = np.array([-1, -11, 0, 4, 5, 3]).astype(ak.uint64)
+            c = c + 2**200
+            d = np.array([-1, -4, 0, 7, 8, 3]).astype(ak.uint64)
+            d = d + 2**200
+        elif dtype == ak.bool:
+            a = np.array([True, False, False, False]).astype(dtype)
+            b = np.array([True, True, False, True]).astype(dtype)
+
+            c = np.array([True, True, False, False]).astype(dtype)
+            d = np.array([True, False, False, False]).astype(dtype)
+        else:
+            a = b = c = d = None
+        return a, b, c, d
+
+    @staticmethod
+    def make_np_arrays_cross_type(dtype):
+        if dtype == ak.int64 or dtype == ak.uint64:
+            a = np.array([-1, 0, 1, 2, 3]).astype(dtype)
+            b = np.array([-1, 0, 9, 2, 7]).astype(dtype)
+        elif dtype == ak.bigint:
+            a = np.array([-1, 0, 1, 2, 3]).astype(ak.uint64)
+            a = a + 2**200
+            b = np.array([-1, 0, 9, 2, 7]).astype(ak.uint64)
+            b = b + 2**200
+        elif dtype == ak.bool:
+            a = np.array([True, False, False, True, True])
+            b = np.array([True, True, False, False, True])
+        else:
+            a = b = None
+        return a, b
+
+    @pytest.mark.parametrize("size", pytest.prob_size)
+    @pytest.mark.parametrize("dtype", INTEGRAL_TYPES)
+    @pytest.mark.parametrize("op", OPS)
+    def test_setops_integral_type(self, size, dtype, op):
+        a, b = self.make_np_arrays(size, dtype)
+
+        func = getattr(ak, op)
+        ak_result = func(ak.array(a, dtype=dtype), ak.array(b, dtype=dtype))
+        np_func = getattr(np, op)
+        np_result = np_func(a, b)
+        assert np.array_equal(ak_result.to_ndarray(), np_result)
+
+        a, b = self.make_np_arrays_small(dtype)
+        func = getattr(ak, op)
+        ak_result = func(ak.array(a, dtype=dtype), ak.array(b, dtype=dtype))
+        np_func = getattr(np, op)
+        np_result = np_func(a, b)
+        assert np.array_equal(ak_result.to_ndarray(), np_result)
```

Comment:
I am leaning away from this just because it feels like parameterizing 2 functions in the same workflow could be confusing. Its straight forward here, but I do not want to get in a habit of oversimplifying/shorten the code to a point where it is unreadable.